src/ast/ast_validations.c

```diff
@@ -1015,14 +1015,22 @@ static AST_Validation _ValidateUnion_Clauses
 		const char *projections[proj_count];
 
 		for(uint j = 0; j < proj_count; j++) {
-			const cypher_astnode_t *proj = cypher_ast_return_get_projection(return_clause, j);
-			const cypher_astnode_t *alias_node = cypher_ast_projection_get_alias(proj);
-			if(alias_node == NULL)  {
-				// The projection was not aliased, so the projection itself must be an identifier.
-				alias_node = cypher_ast_projection_get_expression(proj);
-				ASSERT(cypher_astnode_type(alias_node) == CYPHER_AST_IDENTIFIER);
+			const char *alias = NULL;
+			const cypher_astnode_t *proj =
+				cypher_ast_return_get_projection(return_clause, j);
+			const cypher_astnode_t *alias_node =
+				cypher_ast_projection_get_alias(proj);
+
+			if(alias_node != NULL)  {
+				alias = cypher_ast_identifier_get_name(alias_node);
+			} else {
+				const cypher_astnode_t *exp =
+					cypher_ast_projection_get_expression(proj);
+				cypher_astnode_type_t type = cypher_astnode_type(exp);
+				if(type == CYPHER_AST_IDENTIFIER) {
+					alias = cypher_ast_identifier_get_name(exp);
+				}
```

Comment:
Comment on when we can get here. If this code will be in a function, the comment can be only there - better.

Reply:
OK. The comments are in the new function.